### PR TITLE
WoE: implement Sunk Cost

### PR DESCRIPTION
**Pull request by strib**
Issue: #2907

### CODE REVIEW

File: server/game/cards/06-WoE/SunkCost.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    setupCardAbilities(ability) {
+        this.play({
+            condition: (context) => !!context.player.opponent,
+            target: {
+                mode: 'house',
+                houses: (context) => context.player.opponent.houses
+            },
+            effect: 'make a token creature if {1} randomly discards a card matching {2}',
+            effectArgs: (context) => [context.player.opponent, context.house],
+            effectAlert: true,
```
Comment (romanogit) [inline server/game/cards/06-WoE/SunkCost.js:16]: I do not think `effectAlert` is needed here

File: server/game/cards/06-WoE/SunkCost.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            gameAction: ability.actions.discardAtRandom(),
+            then: (preThenContext) => ({
+                gameAction: ability.actions.conditional({
```
Comment (romanogit) [inline server/game/cards/06-WoE/SunkCost.js:19]: Here you can avoid conditional using amount:
```
ability.actions.makeTokenCreature((context) => ({
   amount: context.preThenEvent.cards.length > 0 &&
                        context.preThenEvent.cards[0].hasHouse(preThenContext.house) ? 1 : 0
})
```

Comment (strib) [reply server/game/cards/06-WoE/SunkCost.js]: Done, thanks.

File: server/game/cards/06-WoE/SunkCost.js (hunk 1 above)
Comment (strib) [reply server/game/cards/06-WoE/SunkCost.js:16]: I disagree -- it is the same type of effect as Allusions of Grandeur, which does have the `effectAlert` (I copied this from there).  But if you have a reason to think we should treat Sunk Cost differently I'm happy to change it.

Comment (strib) [reply server/game/cards/06-WoE/SunkCost.js]: Oops sorry, I was confusing this one with Watch Your Step.  You're right, I removed it.

**Pull request merged**